Source/WebCore/page/playstation/ResourceUsageThreadPlayStation.cpp

```diff
@@ -55,17 +57,31 @@ void ResourceUsageThread::platformCollectCPUData(JSC::VM*, ResourceUsageData& da
 
 void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageData& data)
 {
-    // FIXME: Calculate total dirty memory size for ResourceUsageOverlay
-    data.totalDirtySize = 0;
+    auto& categories = data.categories;
+
+    data.totalDirtySize = bmalloc::api::memoryFootprint();
 
     size_t currentGCHeapCapacity = vm->heap.blockBytesAllocated();
     size_t currentGCOwnedExtra = vm->heap.extraMemorySize();
     size_t currentGCOwnedExternal = vm->heap.externalMemorySize();
     RELEASE_ASSERT(currentGCOwnedExternal <= currentGCOwnedExtra);
 
-    data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
-    data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
-    data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+    categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
+    categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
+    categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+
+    showmap::Result<256> result;
+    result.collect();
```

Comment:
Showmap's `collect()` requires several kernel calls and not a cheap operation. You don't need to check that if `isFastMallocEnabled()` is `false`.

Reply:
Sure. Fixed by https://github.com/WebKit/WebKit/pull/12902/commits/0911d72bc2715f5bd58169779170438e137aeaf7